config/v1/types_dns.go

```diff
@@ -74,6 +80,16 @@ type DNSZone struct {
 	Tags map[string]string `json:"tags,omitempty"`
 }
 
+// Upstream defines the schema for associating a base domain with a list of name resolvers.
+type Upstream struct {
+	// domain is the base domain to match to forward name queries.
+	Domain string `json:"domain"`
+	// resolverList is a list of name servers responsible for resolving name queries from Domain.
+	// Each resolver in the list is represented by an IP address or IP:port if the resolver
+	// listens on a UDP port other than 53.
```

Comment:
It would be useful to document the significance of the ordering of this list.  CoreDNS's forward plugin's default policy is random ordering (per https://coredns.io/plugins/forward/), so I suggest documenting that.  If we need to add a parameter to the API to specify a different policy in the future, such as round-robin or each-in-turn, then that would be an additive change to the API.